hack/aws-acceptance-test-cleanup/main.go

```diff
@@ -759,6 +764,111 @@ func destroyBackoff(ctx context.Context, resourceKind string, resourceID string,
+	}
+
+	if len(rolesToDelete) == 0 {
+		fmt.Println("Found no iamRoles to clean up")
+		return nil
+	}
+
+	// Delete filtered roles
+	for _, role := range rolesToDelete {
+		roleName := aws.StringValue(role.RoleName)
+		err := detachRolePolicies(iamClient, role.RoleName)
+		if err != nil {
+			fmt.Printf("Failed to detach policies for role %s: %v", roleName, err)
+			continue
+		}
+
+		_, err = iamClient.DeleteRole(&iam.DeleteRoleInput{
+			RoleName: role.RoleName,
+		})
+		if err != nil {
+			fmt.Printf("Failed to delete role %s: %v", roleName, err)
+		} else {
+			fmt.Printf("Deleted role: %s", roleName)
+		}
+	}
+
+	return nil
+}
+
+// filterRolesWithPrefix is a callback function used with ListRolesPages.
+// It filters roles based on specified prefix and appends matching roles to rolesToDelete.
+//
+// Parameters:
+// - page: A single page of IAM roles returned by the AWS API
+// - lastPage: A boolean indicating whether this is the last page of results
+// - rolesToDelete: A pointer to the slice where matching roles are accumulated
+// - rolePrefix: The prefix to filter roles by
+func filterRolesWithPrefix(page *iam.ListRolesOutput, lastPage bool, rolesToDelete *[]*iam.Role, rolePrefix string) bool {
+	for _, role := range page.Roles {
+		roleName := aws.StringValue(role.RoleName)
+		if strings.HasPrefix(roleName, rolePrefix) {
+			*rolesToDelete = append(*rolesToDelete, role)
+		}
+	}
+	// Indicates whether to continue pagination (true) or stop (false) until the last page is reached.
+	// When lastPage is true, returning false stops the pagination.
+	return !lastPage
+}
+
+func detachRolePolicies(iamClient *iam.IAM, roleName *string) error {
+	if roleName == nil {
+		return fmt.Errorf("roleName is nil")
+	}
+
+	err := iamClient.ListAttachedRolePoliciesPages(&iam.ListAttachedRolePoliciesInput{
+		RoleName: roleName,
+	}, func(page *iam.ListAttachedRolePoliciesOutput, lastPage bool) bool {
+		err := detachPoliciesFromRole(iamClient, roleName, page)
+		if err != nil {
+			fmt.Printf("Error detaching policies from role %s: %v", *roleName, err)
+		}
+		return !lastPage
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// detachPoliciesFromRole detaches all policies from a given role.
+//
+// Parameters:
+// - iamClient: The IAM service client
+// - roleName: Pointer to the name of the role
+// - page: A single page of attached policies returned by the AWS API
+func detachPoliciesFromRole(iamClient *iam.IAM, roleName *string, page *iam.ListAttachedRolePoliciesOutput) error {
+	for _, policy := range page.AttachedPolicies {
+		if policy.PolicyArn == nil {
+			fmt.Printf("Warning: PolicyArn is nil for a policy attached to: %s", *roleName)
+			continue
+		}
+
+		_, err := iamClient.DetachRolePolicy(&iam.DetachRolePolicyInput{
+			RoleName:  roleName,
+			PolicyArn: policy.PolicyArn,
+		})
+		if err != nil {
+			fmt.Printf("Failed to detach policy %s from role %s: %v\n", aws.StringValue(policy.PolicyArn), *roleName, err)
```

Comment:
Sorry, since we are also de-ref `roleName`, do we also need the `aws.StringValue` to wrap it? The same applies to line 835.